### PR TITLE
Refactorization of Taylor Approximations

### PR DESCRIPTION
**Pull request by nguigs**
This continues #849 to factorize Taylor approximations in the Hyperbolic space. 
- Unfortunately this doesn't improve the numerical accuracy with Pytorch and Tensorflow. 
- Also addresses the #699.
- Avoid recomputing the log in the Frechet mean gradient descent

### CODE REVIEW

File: geomstats/geometry/hyperboloid.py [hunk 1]
```diff
@@ -330,43 +300,19 @@ def exp(self, tangent_vec, base_point):
...
-            1. + SINH_TAYLOR_COEFFS[3] * norm_tangent_vec ** 2
-            + SINH_TAYLOR_COEFFS[5] * norm_tangent_vec ** 4
-            + SINH_TAYLOR_COEFFS[7] * norm_tangent_vec ** 6
-            + SINH_TAYLOR_COEFFS[9] * norm_tangent_vec ** 8)
-        # This avoids dividing by 0.
-        norm_tangent_vec += mask_0_float * 1.0
-        coef_1 += mask_else_float * (gs.cosh(norm_tangent_vec))
-        coef_2 += mask_else_float * (
-            (gs.sinh(norm_tangent_vec) / (norm_tangent_vec)))
+
```
Comment (ninamiolane) [inline geomstats/geometry/hyperboloid.py:303]: Nice! 🎉

**Review by ninamiolane — APPROVED** [review]
LGTM! That cleans the code so much, this is great 🎉

**Comment by ninamiolane** [comment]
@nguigs but it's decreasing coverage, any idea why?

**Comment by nguigs** [comment]
> @nguigs but it's decreasing coverage, any idea why?

Not really, I added tests for the file hyperbolic.
